### PR TITLE
update to log4j2

### PR DESCRIPTION
**Pull request by taba90**
- update dependencies to log4j2.
- migrate log4j.properties to log4j2.properties.
- migrate java code to new log4j2 api.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,173 +4,321 @@
...
       <scope>compile</scope>
     </dependency>
     <dependency>
-      <groupId>com.codahale.metrics</groupId>
-      <artifactId>metrics-log4j</artifactId>
-      <version>3.0.2</version>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-log4j2</artifactId>
+      <scope>compile</scope>
+    </dependency>
```
Comment (aaime) [inline pom.xml:313]: So I guess this is the reason for the upgrade to the newer version.
Did you manually metrics are still properly collected?

File: src/main/webapp/WEB-INF/classes/log4j2.properties [hunk 2]
```diff
@@ -0,0 +1,23 @@
+rootLogger.level = INFO
```
Comment (aaime) [inline src/main/webapp/WEB-INF/classes/log4j2.properties:1]: Same as in the other review, did you manually check more messages are being logged when switching log level?

**Review by aaime — APPROVED** [review]
Looks overall good, see comments below.

File: pom.xml (hunk 1 above)
Comment (taba90) [reply pom.xml:313]: yes tested see my comment below.

File: src/main/webapp/WEB-INF/classes/log4j2.properties (hunk 2 above)
Comment (taba90) [reply src/main/webapp/WEB-INF/classes/log4j2.properties:1]: yes tested see my comment below.

**Comment by taba90** [comment]
thanks for your review @aaime.
This is the console log for the mapfish print lib:
mapfish-log

This is a response of the metrics servlet when enabled to shows metrics about log (using the  dependency on which you commented):

metrics-servlet

**Pull request merged**